Source/DafnyLanguageServer/Handlers/DafnyCodeActionHandler.cs

```diff
@@ -71,18 +71,59 @@ public record DafnyCodeActionWithId(DafnyCodeAction DafnyCodeAction, int Id);
       CommandOrCodeAction t = new CodeAction {
         Title = fixWithId.DafnyCodeAction.Title,
         Data = new JArray(documentUri, fixWithId.Id),
-        Diagnostics = fixWithId.DafnyCodeAction.Diagnostics
+        Diagnostics = fixWithId.DafnyCodeAction.Diagnostics,
+        Kind = CodeActionKind.QuickFix
       };
       return t;
     }
     ).ToArray();
     return new CommandOrCodeActionContainer(codeActions);
   }
 
+
+  public class ErrorMessageCodeActionProvider : DiagnosticDafnyCodeActionProvider {
```

Comment:
Can you move this class to a new file in the same folder as the class `VerificationDafnyCodeActionProvider`? I think it's the language/ folder
Also, please consider migrating all of the lone class DafnyCodeActions inside this handler.